### ISSUE TITLE
Nested [@triggerName] fires when parent trigger is executed. Animation executes before its turn

### ISSUE THREAD

**Issue by mattiLeBlanc** [angular/angular#20811]
## I'm submitting a...

<pre><code>
[ X ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
The issue in a nutshell: A nested @trigger animates when the parent animates, ignoring it's delay.

In the following SVG ( a lock), I have added animated parts to move it's elements and change the colour during the animation.
Right now, when the top lock part close (moves down) the child animation on the path fires too, while I have programmed it to wait 900ms before executing.

![dec-06-2017 09-54-46](https://user-images.githubusercontent.com/5285035/33635350-b7385b62-da6b-11e7-9f29-b88bcbc884bc.gif)

As you can see in the GIF, the top lock part turns red and then the rest of the lock turns red later. They all use the same `@fillState` trigger, however the top part runs at the same time as it @lockState is executed. 
It needs to 1) close the lock, 2) and then change colour. Not at the same time.
This was working in Angular 4.4.2

```
<svg class="icon-lock-open" viewBox="0 0 46 58" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
        <defs>
            <polygon id="path-1" points="0.412 0.329 29.234 0.329 29.234 25.9964355 0.412 25.9964355"></polygon>
            <polygon id="path-3" points="45.762 15.941 45.762 31.846 0 31.846 0 0.036 45.762 0.036"></polygon>
        </defs>
        <g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
          <g>
            <g transform="translate(7.830000, 0.000000)" [@lockState]="state">
              <path  id="lock-top" fill="#38B872" [@fillState]="state"  fill-rule="nonzero" d="M6.1862"></path>
            </g>
            <path id="lock-base-overlay" fill="#E8E9E8" [@lockFillState]="state" fill-rule="nonzero" d="M27.02"></path>
            <g transform="translate(0.000000, 25.230000)">
                <path id="lock-base" fill="#38B872" [@fillState]="state" fill-rule="nonzero" d="M3.255,"></path>
            </g>
            <path id="lock-key-hole" fill="#38B87`2" [@fillState]="state" fill-rule="nonzero" d="M26.8"></path>
          </g>
        </g>
 </svg>
```

```
  animations: [
    trigger( 'lockState', [
      state( 'open', style( {
        transform: 'translate3d( 7.83px, 0, 0)',
        // transform: 'scale(1)',
      } ) ),
      state( 'locked', style( {
        transform: 'translate3d( 7.83px, 5px, 0)',
      } ) ),
      transition( 'open => locked', animate( '100ms 500ms linear' ) ),
      // transition( 'locked => open', animate( '100ms ease-out' ) ),
    ] ),
    // lock part that turns from green to white
    //
    trigger( 'fillState', [
      state( 'open', style( {
        fill: '#38B872',
      } ) ),
      state( 'locked', style( {
        fill: 'red',

      } ) ),
      transition( 'open => locked', animate( '300ms 900ms linear' ) ),
    ] ),
   .... ]
```

## Expected behavior
The @lockState trigger animates first, closing the lock, then the @fillState animation executes (since it is delayed by 900ms). The lock turns from green into red (normally it would be white colour, just for testing).

## Minimal reproduction of the problem with instructions

I will try adding a plunker later.

## What is the motivation / use case for changing the behavior?
Well, I had the animation working in Angular 4.4.2. Now that I have migrated to Angular 5.0.5 the animation is working differently. I can try to create separate @triggerStates or play with groups, but I want to make sure that what I am seeing is not a bug.

## Environment

<pre><code>
Angular version: 5.0.5

Browser:
- [ X ] Chrome (desktop) version 62.0.3202.94
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by matsko** [comment]
@mattiLeBlanc please add a stackblitz demo so I can investigate.

**Comment by mattiLeBlanc** [comment]
Wow, StackBlitz is very nice!

https://stackblitz.com/edit/angular-l2ucdl

I couldn't get my svg-round-progressbar component (third party) to work, getting NG-zone issues.
So I replaced it for a static circle image.
After the pulse effect, the lock closes and it immediately changes color which it shouldn't for another 400ms.

It could be that I might did something wrong, but in my defence: this was working in Angular 4.4.2.

**Comment by visualmatters** [comment]
had a similar ussue here, with a parent component that has an animation attached, and a child component with a different animation; nested inside of it. Parent animation sometimes fires, sometimes not. Removing the animation in the child component fixes it.

**Comment by mattiLeBlanc** [comment]
Yeah I tackled the issue by using the animation group option and using one
state. Made more sense anyway.

**Comment by jamespearce2006** [comment]
Having the same issues with a project we've upgraded from Angular 4.x to 5.2

Several of our animations are now broken and look very ugly - quite frustrating. In our case, most of the animations are entry/leave animations for a component, with different triggers for child items. I'm not sure what (if any) the workaround is.

**Comment by George35mk** [comment]
I have the same problem. any updates ??

**Comment by jfftck** [comment]
I am having an issue that is like this, but when inspecting the DOM I find an additional empty element is created for any element with *ngIf.

**Comment by DavyDeDurpel** [comment]
I think a workaround is to make the names of your states unique across all triggers.

**Comment by bartoszbobin** [comment]
Unfortunately problem still exists in Angular v7.1

making trigger and states name unique dosen't help

**Comment by dmantelli** [comment]
Have the same problem. Child triggers fires when the parent ends, but without animation :(. I'm using Angular 7.2.4 and the problem persists.

**Comment by sasos90** [comment]
Same problem with Angular 8.2.4.

**Comment by mattiLeBlanc** [comment]
@thePunderWoman 😂 this bug was forgotten for many years?? I assume it is no longer relevant, no.

**Comment by JeanMeche** [comment]
We decided to deprecate the animation package in 20.2 in favor of a more lightweight solution that relies on modern CSS : Have a look at https://angular.dev/guide/animations/enter-and-leave

**Comment by thePunderWoman** [comment]
This issue was closed as obsolete due to the introduction of the new `animate.enter` and `animate.leave` APIs. The `@angular/animations` package is now deprecated and will be removed in Angular v23. We hope you give the new APIs a try.

**Comment by mattiLeBlanc** [comment]
@thePunderWoman I mentioned in https://github.com/angular/angular/issues/63595, I can't get the simple example mentioned on this page to work. It looks like the directive animate.enter is not executing.

OKAY: FIXED: I was running Angular 20.1.6 and upgraded tot 20.3.x. 
Sorry about that, but I didnt expect a new feature to introduce via a Minor update of Angular.
